mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1307,20 +1306,25 @@ def predict_with_latency(X_copy):
                 )
 
         X_copy = self.X.copy_to_avoid_mutation()
-        if compute_latency:
-            model_predictions = predict_with_latency(X_copy)
-        else:
-            if self.model is not None:
-                model_predictions = self.model.predict(X_copy)
+        if self.model is not None:
+            if compute_latency:
+                model_predictions = predict_with_latency(X_copy)
             else:
-                if self.dataset.predictions_data is None:
-                    raise MlflowException(
-                        message="Predictions data is missing when model is not provided. "
-                        "Please provide predictions data in the pandas dataset or provide "
-                        "a model.",
-                        error_code=INVALID_PARAMETER_VALUE,
-                    )
-                model_predictions = self.dataset.predictions_data
+                model_predictions = self.model.predict(X_copy)
+        else:
+            if self.dataset.predictions_data is None:
+                raise MlflowException(
+                    message="Predictions data is missing when model is not provided. "
+                    "Please provide predictions data in the pandas dataset or provide "
+                    "a model.",
```

Comment:
super nit:
Could we specify exactly what they should be providing in the pandas dataset? I think my details could be wrong, but along these ideas.
(i.e. `data.predictions` is not defined. It should be defined when using a dataset, or you can provide a model)

Reply:
There are 2 ways to specify the predictions data depending on whether they use mlflow pandas dataset or pandas dataframe. I'll mention:
```
"See the documentation for mlflow.evaluate() for how to specify "
"the predictions data in a dataset."
```